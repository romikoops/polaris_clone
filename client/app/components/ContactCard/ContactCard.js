import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styles from './ContactCard.scss';
import { v4 } from 'node-uuid';
import { Checkbox } from '../Checkbox/Checkbox';
import Truncate from 'react-truncate';

export class ContactCard extends Component {
    constructor(props) {
        super(props);
        this.state = {
            selected: false
        };
        this.toggleSelected = this.toggleSelected.bind(this);
        this.selectContact = this.selectContact.bind(this);
    }
    toggleSelected() {
        const { contactData, toggleSelect } = this.props;
        this.setState({ selected: !this.state.selected });
        toggleSelect(contactData);
    }
    selectContact() {
        const { contactData, select, target, list } = this.props;
        list ? '' : select(target, contactData);
    }
    render() {
        const { contactData, list, theme } = this.props;
        const { contact, location } = contactData;
        const iconStyle = {
            background:
                theme && theme.colors
                    ? '-webkit-linear-gradient(left, ' +
                      theme.colors.primary +
                      ',' +
                      theme.colors.secondary +
                      ')'
                    : 'black',
            paddingRight: '5px'
        };
        return (
            <div
                key={v4()}
                className={`flex-100 layout-row ${styles.contact_card}`}
                onClick={this.selectContact}
            >
                <div className="flex layout-row layout-wrap">
                    <div className="flex-100 layout-row layout-align-space-between-center">
                        <div className="flex-60 layout-row layout-align-start-start layout-wrap">
                            <div className="flex-100 layout-row alyout-align-start-center">
                                <i
                                    className="fa fa-user-circle-o flex-none"
                                    style={iconStyle}
                                />
                                <p className={`flex ${styles.contact_header}`}>
                                    {' '}
                                    {contact.first_name} {contact.last_name}{' '}
                                </p>
                            </div>
                        </div>
                        <div
                            className={`flex-40 layout-row layout-wrap layout-align-start-center ${
                                styles.contact_details
                            }`}
                        >
                            <div className="flex-100 layout-row layout-align-start-center">
                                <i
                                    className="fa fa-envelope flex-none"
                                    style={iconStyle}
                                />
                                <p className="flex-none"> {contact.email} </p>
                            </div>
                        </div>
                    </div>
                    <div className="flex-100 layout-row layout-align-space-between-center">
                        <div className="flex-60 layout-row layout-align-start-start layout-wrap">
                            <div className="flex-100 layout-row alyout-align-start-center">
                                <i
                                    className="fa fa-building-o flex-none"
                                    style={iconStyle}
                                />
                                <p className={`flex ${styles.contact_header}`}>
                                    {' '}
                                    <Truncate lines={1} >{contact.company_name} </Truncate>{}{' '}
                                </p>
                            </div>
                        </div>
                        <div
                            className={`flex-40 layout-row layout-wrap layout-align-start-center ${
                                styles.contact_details
                            }`}
                        >
                             <div className="flex-100 layout-row layout-align-start-center">
                                <i
                                    className="fa fa-phone flex-none"
                                    style={iconStyle}
                                />
                                <p className="flex-none"> {contact.phone} </p>
                            </div>
                        </div>
                    </div>
                    { location && location.geocoded_address ?
                        <div className="flex-100 layout-row layout-align-start-center">
                            <i
                                className="fa fa-globe flex-none"
                                style={iconStyle}
                            />
                            <p className="flex-100"> {location.geocoded_address}</p>
                        </div> :
                        <div className="flex-100" style={{height: '15px'}}></div>
                    }
                </div>
                {list ? (
                    <div className="flex-15 layout-row layout-align-center-center">
                        <Checkbox
                            onChange={this.toggleSelected}
                            checked={this.state.selected}
                        />
                    </div>
                ) : (
                    ''
                )}
            </div>
        );
    }
}

ContactCard.propTypes = {
    contactData: PropTypes.array,
    theme: PropTypes.object,
    select: PropTypes.func,
    list: PropTypes.bool,
    toggleSelect: PropTypes.func,
    target: PropTypes.string
};
