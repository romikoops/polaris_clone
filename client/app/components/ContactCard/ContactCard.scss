:local {
  .contact_header{
    margin: 3px 0;
    color: #212121; 
    font-size: 16px;  
    line-height: 18px;
  }
  .contact_details p{
    margin: 3px 0;
    color: #212121; 
    font-size: 14px;  
    line-height: 16px;
  }
  .contact_card {
    height: 113px !important;
    margin: 5px;
    background-color: #F2F2F2;  
    padding: 15px 15px 0px 15px;
    border-radius: 2px;
    cursor: pointer;
    position: relative;
    .overlay {
      opacity: 0;
      background: black;
      position: absolute;
      top: 0;
      bottom: 0;
      left: 0;
      right: 0;
      transition: all 0.3s ease;
    }
    &:hover .overlay {
      opacity: 0.03;
    }
  }
}