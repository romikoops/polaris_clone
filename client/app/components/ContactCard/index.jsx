import React from 'react'
import Truncate from 'react-truncate'
import PropTypes from '../../prop-types'
import styles from './ContactCard.scss'
import { gradientTextGenerator } from '../../helpers'

export default function ContactCard ({
  contactData, theme, contactType, select
}) {
  const { contact, location } = contactData
  const iconStyle = {
    ...gradientTextGenerator(theme.colors.primary, theme.colors.secondary),
    width: '28px',
    padding: '3px 0'
  }
  const fullAddress = (
    <p className="flex-100" style={{ margin: 0 }}>
      {location.street ? `${location.street}, ` : ''}
      {location.streetNumber ? `${location.streetNumber}, ` : ''}
      {location.zipCode ? `${location.zipCode}, ` : ''}
      {location.city ? `${location.city}` : ''}
      {location.country ? ', ' : ''}
      {location.country ? `${location.country}, ` : ''}
    </p>
  )

  return (
    <div
      className={`flex-100 layout-row ${styles.contact_card}`}
      onClick={() => select(contactData, contactType)}
    >
      <div className={styles.overlay} />
      <div className="flex layout-row layout-wrap">
        <div className="flex-100 layout-row layout-align-space-between-center">
          <div className="flex-60 layout-row layout-align-start-start layout-wrap">
            <div className="flex-100 layout-row alyout-align-start-center">
              <i className="fa fa-user-circle-o flex-none clip" style={iconStyle} />
              <p className={`flex ${styles.contact_header}`}>
                {contact.firstName} {contact.lastName}
              </p>
            </div>
          </div>
          <div className={
            `flex-40 layout-row layout-wrap layout-align-start-center ${styles.contact_details}`
          }
          >
            <div className="flex-100 layout-row layout-align-start-center">
              <i className="fa fa-envelope flex-none clip" style={iconStyle} />
              <p className="flex-none"> {contact.email} </p>
            </div>
          </div>
        </div>
        <div className="flex-100 layout-row layout-align-space-between-center">
          <div className="flex-60 layout-row layout-align-start-start layout-wrap">
            <div className="flex-100 layout-row layout-align-start-center">
              <i className="fa fa-building-o flex-none clip" style={iconStyle} />
              <p className={`flex-80 ${styles.contact_header}`}>
                <Truncate trimWhitespace> {contact.companyName} </Truncate>
              </p>
            </div>
          </div>
          <div className={
            `flex-40 layout-row layout-wrap layout-align-start-center ${styles.contact_details}`
          }
          >
            <div className="flex-100 layout-row layout-align-start-center">
              <i className="fa fa-phone flex-none clip" style={iconStyle} />
              <p className="flex-none"> {contact.phone} </p>
            </div>
          </div>
        </div>
        <div className="flex-100 layout-row layout-align-start-center">
          <i className="fa fa-globe flex-none clip" style={iconStyle} />
          <p className="flex-100">
            {(location && location.geocodedAddress) || fullAddress}
          </p>
        </div>
      </div>
    </div>
  )
}

ContactCard.propTypes = {
  contactData: PropTypes.shape({
    contact: PropTypes.object,
    location: PropTypes.object
  }).isRequired,
  theme: PropTypes.theme,
  select: PropTypes.func.isRequired,
  contactType: PropTypes.string
}

ContactCard.defaultProps = {
  theme: null,
  contactType: ''
}
