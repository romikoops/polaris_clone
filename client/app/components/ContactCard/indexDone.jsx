import React from 'react'
import Truncate from 'react-truncate'
import PropTypes from '../../prop-types'
import styles from './ContactCard.scss'
import { gradientTextGenerator } from '../../helpers'
import { ROW, trim, WRAP_ROW, ALIGN_START } from '../../classNames'

const CONTAINER = `CONTACT_CARD flex-100 layout-row ${styles.contact_card}`
const BUILDING_ICON = 'fa fa-building-o flex-none clip'
const ENVELOPE_ICON = 'fa fa-envelope flex-none clip'
const GLOBE_ICON = 'fa fa-globe flex-none clip'
const PHONE_ICON = 'fa fa-phone flex-none clip'
const USER_CIRCLE_ICON = 'fa fa-user-circle-o flex-none clip'

export default function ContactCard ({
  contactData, theme, contactType, select
}) {
  const { contact, location } = contactData
  const iconStyle = {
    ...gradientTextGenerator(theme.colors.primary, theme.colors.secondary),
    width: '28px',
    padding: '3px 0'
  }

  return (
    <div
      className={CONTAINER}
      onClick={() => select(contactData, contactType)}
    >
      <div className={styles.overlay} />

      <div className={`flex ${WRAP_ROW()}`}>
        <div className={`${ROW(100)} layout-align-space-between-center`}>
          <div className={`${WRAP_ROW(60)} ${ALIGN_START}`}>
            <div className={`${ROW(100)} layout-align-start-center`}>
              <i className={USER_CIRCLE_ICON} style={iconStyle} />
              <p className={`flex ${styles.contact_header}`}>
                {contact.firstName} {contact.lastName}
              </p>
            </div>
          </div>

          <div className={`${WRAP_ROW(40)} layout-align-start-center ${styles.contact_details}`}>
            <div className={`${ROW(100)} layout-align-start-center`}>
              <i className={ENVELOPE_ICON} style={iconStyle} />
              <p className="flex-none"> {contact.email} </p>
            </div>
          </div>
        </div>

        <div className={`${ROW(100)} layout-align-space-between-center`}>
          <div className={`${WRAP_ROW(60)} ${ALIGN_START}`}>
            <div className={`${ROW(100)} layout-align-start-center`}>
              <i className={BUILDING_ICON} style={iconStyle} />
              <p className={`flex-80 ${styles.contact_header}`}>
                <Truncate trimWhitespace> {contact.companyName} </Truncate>
              </p>
            </div>
          </div>

          <div className={trim(`
            ${WRAP_ROW(40)} 
            layout-align-start-center 
            ${styles.contact_details}
          `)}
          >
            <div className={`${ROW(100)} layout-align-start-center`}>
              <i className={PHONE_ICON} style={iconStyle} />
              <p className="flex-none"> {contact.phone} </p>
            </div>
          </div>
        </div>

        <div className={`${ROW(100)} layout-align-start-center`}>
          <i className={GLOBE_ICON} style={iconStyle} />
          <p className="flex-100">
            {(location && location.geocodedAddress) || location.fullAddress}
          </p>
        </div>
      </div>
    </div>
  )
}

ContactCard.propTypes = {
  contactData: PropTypes.shape({
    contact: PropTypes.object,
    location: PropTypes.object
  }).isRequired,
  theme: PropTypes.theme,
  select: PropTypes.func.isRequired,
  contactType: PropTypes.string
}

ContactCard.defaultProps = {
  theme: null,
  contactType: ''
}
