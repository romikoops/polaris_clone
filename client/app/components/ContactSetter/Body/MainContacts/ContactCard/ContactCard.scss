:local {
  .contact_card {
    margin: 5px;
    position: relative;
    border: 2px solid black;
    height: 242px !important;
    padding: 25px 20px;
  	i {
		  -webkit-background-clip: text !important;
			-webkit-text-fill-color: transparent !important;

  	}
  	i.main_icon {
		  font-size: 24px;
		  padding: 0 10px;
		  width: 19px;
    }
    i.plus_icon {
      background: rgb(140, 140, 140);
			@media screen and (-ms-high-contrast: active),
			(-ms-high-contrast: none) {
				color: grey;
				background: transparent;
			}
      position: absolute;
      right: 15px;
      top: 15px;
      font-size: 22px;
      transition: all 0.3s ease;
      opacity: 0.7;
      cursor: pointer;
      &:hover {
        opacity: 1;
      }
    }
    i.edit_icon {
			background: rgb(140, 140, 140);
			@media screen and (-ms-high-contrast: active),
			(-ms-high-contrast: none) {
				color: grey;
				background: transparent;
			}
      position: absolute;
      right: 46px;
      top: 15px;
      font-size: 22px;
      transition: all 0.3s ease;
      opacity: 0.7;
      cursor: pointer;
      &:hover {
        opacity: 1;
      }
    }
	  h3.contact_name {
	  	font-size: 26px;
	  	margin: 0 0 25px 0;
	  }
		.secondary_info {
			line-height: 26px;
			margin: 0;
			font-size: 19px;
			font-weight: normal;
		}
		.secondary_info_city {
			line-height: 22px;
			margin: 0;
			font-size: 16px;
			font-weight: normal;
		}
		.secondary_info_country {
			line-height: 18px;
			margin: 0;
			font-size: 14px;
			font-weight: normal;
		}
		.contact_data_sec {
			padding: 20px 5px;
			i {
				width: unset;
				padding: 0px 5px 0 0;
				font-size: 12px;
			}
			.contact_data {
				font-size: 12px;
				margin: 0;
			}
		}
  }
}
