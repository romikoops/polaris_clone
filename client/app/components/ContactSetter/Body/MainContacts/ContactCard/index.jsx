import React from 'react'
import Truncate from 'react-truncate'
import PropTypes from '../../../../../prop-types'
import styles from './ContactCard.scss'
import { gradientTextGenerator } from '../../../../../helpers'

function commaSeparatedWhenBothExist (str1, str2) {
  return str1 && str2 ? `${str1}, ${str2}` : (str1 || str2)
}
function locationForDisplay (location) {
  const {
    street, streetNumber, city, country, zipCode
  } = location
  const cityCountry = commaSeparatedWhenBothExist(city, country)

  const streetWithNumber = street && street + (streetNumber && ` ${streetNumber}`)
  const addressDetails = commaSeparatedWhenBothExist(streetWithNumber, zipCode)

  return { cityCountry, addressDetails }
}

export default function ShipmentContactsBoxMainContactsContactCard ({
  contactData, theme, contactType, showAddressBook
}) {
  const { contact, location } = contactData
  const iconStyle = {
    ...gradientTextGenerator(theme.colors.primary, theme.colors.secondary)
  }

  const borderStyles = theme ? { borderColor: theme.colors.secondary } : {}

  const editIcon = (
    <i
      className={`${styles.edit_icon} fa fa-pencil-square-o`}
      onClick={() => showAddressBook(contactType)}
    />
  )

  const { addressDetails } = locationForDisplay(location)

  return (
    <div className={`flex-100 layout-row layout-wrap ${styles.contact_card}`} style={borderStyles}>
      {editIcon}
      <div className="flex-100 layout-row layout-align-start-start">
        <i className={`${styles.main_icon} fa fa-user`} style={iconStyle} />
        <h3 className={`${styles.contact_name}`}>
          {contact.firstName} {contact.lastName} <br />
          <span className={styles.secondary_info}>
            <Truncate lines={1}>{contact.companyName} </Truncate>
          </span>
        </h3>
      </div>
      <div className="flex-100 layout-row layout-align-start-start layout-wrap">
        <div className="flex-100 layout-row layout-align-space-around-center">
          <i className={`${styles.main_icon} fa fa-map-marker flex-10`} style={iconStyle} />
          <p className={`${styles.secondary_info} flex`}>
            { addressDetails }
          </p>
        </div>
        <div className="flex-100 layout-row layout-align-end-center">
          <p className={`${styles.secondary_info_city} flex-90 offset-10`}>
            <b> { location.city } </b>
          </p>
        </div>
        <div className="flex-100 layout-row layout-align-end-center">
          <p className={`${styles.secondary_info_country} flex-90 offset-10`}>
            <Truncate lines={1}> { location.country }</Truncate>
          </p>
        </div>
      </div>
      <div className={
        `${styles.contact_data_sec} flex-100 ` +
        'layout-row layout-wrap layout-align-center-center'
      }
      >
        <div className="flex-none layout-row layout-align-start-center">
          <i className="fa fa-envelope flex-none" style={iconStyle} />
          <p className={styles.contact_data}> {contact.email} </p>
        </div>
        <div className="offset-5 flex-none layout-row layout-align-start-center">
          <i className="fa fa-phone flex-none" style={iconStyle} />
          <p className={styles.contact_data}> {contact.phone} </p>
        </div>
      </div>
    </div>
  )
}

ShipmentContactsBoxMainContactsContactCard.propTypes = {
  contactData: PropTypes.shape({
    contact: PropTypes.object,
    location: PropTypes.object
  }).isRequired,
  theme: PropTypes.theme,
  contactType: PropTypes.string,
  showAddressBook: PropTypes.func
}

ShipmentContactsBoxMainContactsContactCard.defaultProps = {
  theme: null,
  contactType: '',
  showAddressBook: null
}
