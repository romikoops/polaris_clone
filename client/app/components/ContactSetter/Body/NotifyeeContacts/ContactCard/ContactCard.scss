:local {
	.contact_card {
		background: #F2F2F2;
		padding: 20px;
		position: relative;
		h3.contact_name {
			margin: 0;
			font-size: 18px;
			span.secondary_info {
				font-size: 14px;
				font-weight: normal;
			}
		}
		i.user_icon {
			margin-right: 12px;
			padding: 0 2px;
    }
    .icons {
      i {
        position: absolute;
        font-size: 22px;
        opacity: 0.3;
        transition: all 0.3s ease;
        &:hover {
          opacity: 0.5;
        }
      }
      i.remove_icon {

        right: 30px;

      }
      i.edit_icon {
        right: 70px;
      }
    }
	}
}
