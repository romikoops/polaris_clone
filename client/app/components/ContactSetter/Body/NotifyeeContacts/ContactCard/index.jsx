import React from 'react'
import Truncate from 'react-truncate'
import { v4 } from 'node-uuid'
import PropTypes from '../../../../../prop-types'
import styles from './ContactCard.scss'
import { gradientTextGenerator } from '../../../../../helpers'

export default function ContactSetterBodyNotifyeeContactsContactCard ({
  theme, contactData, removeFunc
}) {
  const { contact } = contactData
  const iconStyle = {
    ...gradientTextGenerator(theme.colors.primary, theme.colors.secondary)
  }

  const removeIcon = (
    <i
      className={`${styles.remove_icon} fa fa-trash`}
      onClick={() => removeFunc()}
    />
  )

  return (
    <div
      key={v4()}
      className={`layout-row ${styles.contact_card}`}
    >
      { removeIcon }
      <div className="flex-100 layout-row layout-align-start-start">
        <i className={`${styles.user_icon} fa fa-user clip`} style={iconStyle} />
        <h3 className={`${styles.contact_name}`}>
          {contact.firstName} {contact.lastName} <br />
          <span className={styles.secondary_info}>
            <Truncate lines={1}>{contact.companyName} </Truncate>
          </span>
        </h3>
      </div>
    </div>
  )
}

ContactSetterBodyNotifyeeContactsContactCard.propTypes = {
  contactData: PropTypes.shape({
    contact: PropTypes.object,
    location: PropTypes.object
  }).isRequired,
  theme: PropTypes.theme,
  removeFunc: PropTypes.func
}

ContactSetterBodyNotifyeeContactsContactCard.defaultProps = {
  theme: null,
  removeFunc: null
}
