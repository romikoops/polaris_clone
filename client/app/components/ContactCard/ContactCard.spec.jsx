import * as React from 'react'
import { mount, shallow } from 'enzyme'
import { theme, user, identity } from '../../mocks'

jest.mock('react-truncate', () =>
  // eslint-disable-next-line react/prop-types
  ({ children }) => <span>{children}</span>)

// eslint-disable-next-line
import ContactCard from './ContactCard'

const editedUser = {
  ...user,
  firstName: user.first_name,
  lastName: user.last_name
}

const propsBase = {
  contactData: {
    contact: editedUser,
    location: {}
  },
  theme,
  select: identity,
  contactType: '',
  removeFunc: null,
  popOutHover: false
}

let wrapper

const createWrapper = propsInput => mount(<ContactCard {...propsInput} />)

beforeEach(() => {
  wrapper = createWrapper(propsBase)
})

test('shallow render', () => {
  expect(shallow(<ContactCard {...propsBase} />)).toMatchSnapshot()
})

test('props.select is called upon click', () => {
  const props = {
    ...propsBase,
    select: jest.fn()
  }
  const dom = createWrapper(props)
  const clickableDiv = dom.find('div.contact_card').first()

  expect(props.select).not.toHaveBeenCalled()
  clickableDiv.simulate('click')
  expect(props.select).toHaveBeenCalled()
})

test('remove icon is visible when passing props.removeFunc', () => {
  const props = {
    ...propsBase,
    removeFunc: jest.fn()
  }
  const dom = createWrapper(props)

  const removeDefault = wrapper.find('.contact_card > i')
  const remove = dom.find('.contact_card > i')

  expect(removeDefault).toHaveLength(0)
  expect(remove).toHaveLength(1)
})

test('click on remove icon calls props.removeFunc', () => {
  const props = {
    ...propsBase,
    removeFunc: jest.fn()
  }
  const dom = createWrapper(props)

  const remove = dom.find('.contact_card > i').first()

  expect(props.removeFunc).not.toHaveBeenCalled()
  remove.simulate('click')
  expect(props.removeFunc).toHaveBeenCalled()
})

test('correctly render contact details', () => {
  const name = wrapper.find('.contact_header').first().text()
  const email = wrapper.find('.contact_details p').first().text()
  const phone = wrapper.find('.contact_details p').at(1).text()

  expect(name.includes(user.first_name)).toBeTruthy()
  expect(name.includes(user.last_name)).toBeTruthy()
  expect(email.includes(user.email)).toBeTruthy()
  expect(phone.includes(user.phone)).toBeTruthy()
})

test('location', () => {
  const locationProp = {
    geocodedAddress: 'FOO_GEO',
    fullAddress: 'FOO_FULL_ADDRESS'
  }
  const props = {
    ...propsBase,
    contactData: {
      ...propsBase.contactData,
      location: locationProp
    }
  }
  const withLocation = createWrapper(props)

  const defaultLocation = wrapper.find('p.flex-100').last().text()
  const location = withLocation.find('p.flex-100').last().text()

  expect(defaultLocation).toBe('')
  expect(location).toBe(locationProp.geocodedAddress)
})
