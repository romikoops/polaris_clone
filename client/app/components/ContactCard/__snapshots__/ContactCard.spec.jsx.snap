// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow render 1`] = `
<div
  className="flex-100 layout-row contact_card "
  key="8e252bf0-b0b9-4dd8-8cec-420d9a1b1889"
  onClick={[Function]}
>
  <div
    className="flex layout-row layout-wrap"
  >
    <div
      className="flex-100 layout-row layout-align-space-between-center"
    >
      <div
        className="flex-60 layout-row layout-align-start-start layout-wrap"
      >
        <div
          className="flex-100 layout-row alyout-align-start-center"
        >
          <i
            className="fa fa-user-circle-o flex-none"
            style={
              Object {
                "background": "-webkit-linear-gradient(left, #333,#fafafa)",
                "padding": "3px 0",
                "width": "28px",
              }
            }
          />
          <p
            className="flex contact_header"
          >
            John
             
            Doe
          </p>
        </div>
      </div>
      <div
        className="flex-40 layout-row layout-wrap layout-align-start-center contact_details"
      >
        <div
          className="flex-100 layout-row layout-align-start-center"
        >
          <i
            className="fa fa-envelope flex-none"
            style={
              Object {
                "background": "-webkit-linear-gradient(left, #333,#fafafa)",
                "padding": "3px 0",
                "width": "28px",
              }
            }
          />
          <p
            className="flex-none"
          >
             
            foo@bar.baz
             
          </p>
        </div>
      </div>
    </div>
    <div
      className="flex-100 layout-row layout-align-space-between-center"
    >
      <div
        className="flex-60 layout-row layout-align-start-start layout-wrap"
      >
        <div
          className="flex-100 layout-row alyout-align-start-center"
        >
          <i
            className="fa fa-building-o flex-none"
            style={
              Object {
                "background": "-webkit-linear-gradient(left, #333,#fafafa)",
                "padding": "3px 0",
                "width": "28px",
              }
            }
          />
          <p
            className="flex contact_header"
          >
            <Component
              lines={1}
            >
               
            </Component>
          </p>
        </div>
      </div>
      <div
        className="flex-40 layout-row layout-wrap layout-align-start-center contact_details"
      >
        <div
          className="flex-100 layout-row layout-align-start-center"
        >
          <i
            className="fa fa-phone flex-none"
            style={
              Object {
                "background": "-webkit-linear-gradient(left, #333,#fafafa)",
                "padding": "3px 0",
                "width": "28px",
              }
            }
          />
          <p
            className="flex-none"
          >
             
            6345789
             
          </p>
        </div>
      </div>
    </div>
    <div
      className="flex-100 layout-row layout-align-start-center"
    >
      <i
        className="fa fa-globe flex-none"
        style={
          Object {
            "background": "-webkit-linear-gradient(left, #333,#fafafa)",
            "padding": "3px 0",
            "width": "28px",
          }
        }
      />
      <p
        className="flex-100"
      />
    </div>
  </div>
</div>
`;
